### PR TITLE
UTOPIA-209: Rollback Express API in Git Repo

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- [UTOPIA-209](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-209)
- Revert of existing Express API demo app
- Boilerplate NestJSf App

Note: 
This PR only introduces boilerplate NestJS boilerplate app. Any additional configurations like swagger, routing versions, x-powered-by, etc will be taken care in the next ticket [UTOPIA-210: Init NestJS API Project](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-210).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

After running `npm start`, and running localhost:3000 returns `Hello world`

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [x] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [x] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [x] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

### CODE REVIEW

File: src/backend/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all"
+}
```
Comment (akroon3r) [inline src/backend/.prettierrc:4]:
```suggestion
}

```

File: src/backend/test/app.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  beforeEach(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  it('/ (GET)', () => {
```
Comment (akroon3r) [inline src/backend/test/app.e2e-spec.ts:18]: Instead of serving a 200 at root, I'd like this to be /api/healthcheck.

Let me know what you think please.

**Review by akroon3r — CHANGES_REQUESTED** [review]
This is my first pass review. Awesome job on your first commit to the repo Kushal.

Check out my review comments and let me know what you think please. 🚀

File: src/backend/test/app.e2e-spec.ts (hunk 2 above)
Comment (kushal-arora-fw) [reply src/backend/test/app.e2e-spec.ts:18]: Sure, it definitely does! In fact I liked your previously committed idea of api versions and /health changes. 
image

But this is just a boilerplate code to get the ball rolling keeping just the limited pieces of the existing code and unblock @binggaofw as well. I was thinking to make all the configuration changes like swagger, health API changes in the upcoming PR.

Comment (akroon3r) [reply src/backend/test/app.e2e-spec.ts:18]: Love it. Thank you addressing my feedback.

Comment (akroon3r) [reply src/backend/test/app.e2e-spec.ts:18]: Food for thought:

API Guidelines: https://developer.gov.bc.ca/Data-and-APIs/BC-Government-API-Guidelines#api-lifecycle-management 

Under the URI vs Accept Header Versioning heading, the BC Gov guidelines suggest we only should on include /v1 or /vX in our URI if we have previous versions of our API in place. At this moment in time we currently do not.

I'd like to talk about this more, possibly at our next meeting on Monday.

Comment (kushal-arora-fw) [reply src/backend/test/app.e2e-spec.ts:18]: Great insight! Thanks for going through the docs. Let's proceed without it for now, and change to `vx` as and when required. And sure - let's chat for about it on Monday 👍

**Review by binggaofw — APPROVED** [review]
looks good to me

**Comment by kushal-arora-fw** [comment]
@akroon3r - Proceeding with the merge. I guess we are using Merge Commit as our Merge strategy. 

Also - can you please look into the failing workflow - seems like this happens on a followup commit when the users are already added to the PR?

**Pull request merged**